docker/docker.go

```diff
@@ -0,0 +1,72 @@
+package docker
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	ecmHTTP "github.com/rancher/ecm-distro-tools/http"
+
+	"github.com/sirupsen/logrus"
+)
+
+const registryURL = "https://hub.docker.com"
+
+type DockerImage struct {
+	Architecture string    `json:"architecture"`
+	Status       string    `json:"status"`
+	Size         int       `json:"size"`
+	LastPushed   time.Time `json:"last_pushed"`
+}
+
+type DockerTag struct {
+	Name   string        `json:"name"`
+	Images []DockerImage `json:"images"`
+}
+
+// CheckImageArchs returns true if an image exists and has all the provided architectures
+func CheckImageArchs(ctx context.Context, org, repo, tag string, archs []string) (bool, error) {
```

Comment:
I'm not sure we need to return a bool for this function. It appears as though the only time true is returned is when there's no error. And doesn't seem like false is returned when there is no error. With that, I think we can get rid of the boolean value and base the determination off of whether there's an error or not.